extensions/amp-video/0.1/video-cache.js

```diff
@@ -95,3 +104,34 @@ function applySourcesToVideo(videoEl, sources) {
       videoEl.insertBefore(sourceEl, videoEl.firstChild);
     });
 }
+
+/**
+ * Moves the src attribute to a source element to enable playing from multiple
+ * sources: the cached ones and the fallback initial src.
+ * @param {!Element} videoEl
+ * @param {!Window} win
+ * @return {!Promise}
+ */
+function replaceSrcWithSourceElement(videoEl, win) {
+  if (!videoEl.hasAttribute('src')) {
+    return;
+  }
+  const sourceEl = win.document.createElement('source');
+  const srcAttr = videoEl.getAttribute('src');
+  sourceEl.setAttribute('src', srcAttr);
+
+  const typeAttr = videoEl.getAttribute('type');
```

Comment:
Nit: Don't we also need to remove the `type` attribute from the videoEl? Let's say `amp-video[type]` is `mp4` but the video cache returns `wmv`, then the type on `amp-video` could be wrong

Reply:
Let's remove it to keep the DOM clean and updated. Just for the record, since the video would not even get a `src` the browser would not consider it for playback, so it shouldn't have any impact.

Done